### PR TITLE
perf(entity.get): add attribute change tracking

### PR DESCRIPTION
**Pull request by smothers**
- improve re-render performance by ~2x on light pages
- improve re-render performance by ~4x on heavy pages with ~10k components
- only trigger re-renders on attributes that have changed
- instrument `entity.get` to track used attributes and scope them to their respective hook
- update the todo example to better leverage these new performance features
- add performance section to readme

### CODE REVIEW

**Comment by smothers** [comment]
@rubinovitz when you get a chance to review this please also QA it locally. It touches a lot and I'd love another set of eyes on it.

**Comment by rubinovitz** [comment]
@smothers Sure thing!

File: README.md [hunk 1]
```diff
@@ -149,6 +149,37 @@ todos
 .map(todo => todo.get('name'))
 ```
 
+## Performance
+
+Homebase React tracks the attributes consumed in each component via the `entity.get` function and scopes those attributes to their respective `useEntity` or `useQuery` hook. Re-renders are only triggered when an attribute changes.
+
+The default caching reduces unnecessary re-renders and virtual dom thrashing a lot. That said, it is still possible to trigger more re-renders than you might want.
```
Comment (rubinovitz) [inline README.md:156]: nitpick: s/virtual dom/virtual DOM

**Review by rubinovitz — CHANGES_REQUESTED** [review]
@smothers I'm getting "Error: Invalid Hook" call when I try to render the Counter example now.

**Comment by rubinovitz** [comment]
@smothers oh standby I think this is the problem I had last time I tried to test locally and it thinks there's two Reacts. 😬 I'll do the fix I did last time and test again in the morning. You don't need to do anything.

**Comment by rubinovitz** [comment]
@smothers I'm still getting the error after `npm link homebase-react/node_modules/react` , which fixed the error last time it was a double React issue. Do you get an error on your end running this version in a React js app?

**Comment by smothers** [comment]
I'm getting the error if I use yarn, but npm lets me link to the local package

**Review by rubinovitz — APPROVED** [review]
LGTM 🏎

**Pull request merged**